### PR TITLE
(fix) O3-3337 - Support expressions that don't use whitespace as delimiters

### PR DESCRIPTION
**Pull request by arodidev**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR introduces a fix for various failing expression evaluations due to improper formatting during form construction.
## Related Issue

https://openmrs.atlassian.net/browse/O3-3337

### CODE REVIEW

File: src/utils/expression-runner.ts [hunk 1]
```diff
@@ -75,7 +80,7 @@ export function evaluateExpression(
     _,
   };
 
-  expression = linkReferencedFieldValues(fields, fieldValues, parts);
+  expression = expressionFormatter(linkReferencedFieldValues(fields, fieldValues, parts), fields, fieldValues);
```
Comment (pirupius) [inline src/utils/expression-runner.ts:83]: @arodidev based on our conversion a few days ago. Does the splitting remain the same? and the only difference is the expression being resolved regardless if you have spaces or not?

Comment (arodidev) [reply src/utils/expression-runner.ts:83]: @pirupius pius Yes, that is correct. The formatting of the expression remains unaffected, but rather a `formField` namespace is prefixed to the field id.

File: src/utils/expression-parser.test.ts [hunk 2]
```diff
@@ -306,3 +312,36 @@ describe('hasParentheses', () => {
     expect(hasParentheses(expression)).toBe(true);
   });
 });
+
+describe('expression formatter function', () => {
+  it('should replace the field id with fieldValue.field id for all form fields', () => {
```
Comment (brandones) [inline src/utils/expression-parser.test.ts:317]:
```suggestion
describe('expression formatter function', () => {
  it('should not care about whether there are spaces around operators', () => {
```

Comment (brandones) [reply src/utils/expression-parser.test.ts:317]: It's better to be precise about what the purpose of your test is.

**Review by brandones — APPROVED** [review]
LGTM. Next time you get suggestions on your PRs please integrate them rather than just giving a thumbs up.

**Comment by arodidev** [comment]
Hi @brandones, the work on this PR had stalled since @pirupius had picked up the issue, albeit using a different approach.

**Comment by brandones** [comment]
Wait so should we revert this?

When things like that happen, please leave comments on PRs so that people know what's going on.

**Comment by samuelmale** [comment]
@brandones I would say we revert because this is a half-backed solution. Yes it does the necessary interpolation (further hydrates the expression by updating any lurking referenced field's value path) but it doesn't update the relationships (dependencies) etc. The ideal solution is to properly decompose the expression into parts and we were planning on [using a third party library](https://github.com/EricSmekens/jsep).

**Comment by samuelmale** [comment]
>updating the field dependants was not an area of concern while working on this issue since it was non-problematic.

I would say it is a secondary concern.

**Pull request merged**